### PR TITLE
Intersect polyline plane

### PR DESCRIPTION
**Pull request by og76**
This is a very special one
I could make the edges as an option to actually have vectors or vectors+edges, but somehow I prefer it this way

### CODE REVIEW

File: nodes/geometry/intersect_polyline_plane.py [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+    
+    edgesType = EnumProperty(name = "Plane Type", default = "POINTS",
+        items = edgesTypeItems, update = edgesTypeChanged)
+    message = StringProperty(name = "Message", default = "Expecting Points")
+    cyclic = BoolProperty(name = "Cyclic points", description = "Consider last point to first point also", 
+        default = True, update = executionCodeChanged)
+        
+    def create(self):
+        self.newInput("Vector", "Positions", "positions")
+        self.newInput("Edge Indices List", "Edge Indices", "edges")
```
Comment (JacquesLucke) [inline nodes/geometry/intersect_polyline_plane.py]: the convention I use everywhere is:
socket = self.newInput(...)
socket.setting1 = ...
socket.setting2 = ...

**Comment by JacquesLucke** [comment]
this node looks more complicated than the others, I will have to check it in more detail a bit later.

**Comment by og76** [comment]
for examples see the issue #430 

as formula, it does:
- define edges as vecs in order or from edges (for different use, as poly or edges of an obj)
- (pre)check if the edges are cut (very important, see below)
- inters only if the case

the dot  check is important cause:
- it reduces the inters ops to those needed
- it provides position relative to plane normal  - / 0 / + (helps filter if u want to keep only above/below)
- for coplanar, I return the first point as the intx, so if the poly is in the plane it returns original points (the inters func itself does not)

I'll probably make it into an algo later, but for now I'll let it inside the node, with yeld etc

File: nodes/geometry/intersect_polyline_plane.py [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+    bl_label = "Intersect Polyline Plane"
+    bl_width_default = 160
+    
+    def edgesTypeChanged(self, context):
+        executionCodeChanged()
+    
+    edgesType = EnumProperty(name = "Plane Type", default = "POINTS",
+        items = edgesTypeItems, update = edgesTypeChanged)
+    message = StringProperty(name = "Message", default = "Expecting Points")
+    cyclic = BoolProperty(name = "Cyclic points", description = "Consider last point to first point also",
```
Comment (JacquesLucke) [inline nodes/geometry/intersect_polyline_plane.py]: capitalize to "Cyclic Points"

**Comment by og76** [comment]
ok?

**Pull request merged**